core/vtk/ttkTableDistanceMatrix/ttkTableDistanceMatrix.cpp

```diff
@@ -0,0 +1,122 @@
+#include <ttkTableDistanceMatrix.h>
+#include <ttkUtils.h>
+
+#include <vtkDataSet.h>
+#include <vtkDoubleArray.h>
+#include <vtkInformation.h>
+#include <vtkInformationVector.h>
+#include <vtkNew.h>
+#include <vtkObjectFactory.h>
+#include <vtkTable.h>
+
+#include <regex>
+
+vtkStandardNewMacro(ttkTableDistanceMatrix);
+
+ttkTableDistanceMatrix::ttkTableDistanceMatrix() {
+  SetNumberOfInputPorts(1);
+  SetNumberOfOutputPorts(1);
+}
+
+int ttkTableDistanceMatrix::FillInputPortInformation(int port,
+                                                     vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkTable");
+    return 1;
+  }
+  return 0;
+}
+
+int ttkTableDistanceMatrix::FillOutputPortInformation(int port,
+                                                      vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkDataObject::DATA_TYPE_NAME(), "vtkTable");
+    return 1;
+  }
+  return 0;
+}
+
+int ttkTableDistanceMatrix::RequestData(vtkInformation * /*request*/,
+                                        vtkInformationVector **inputVector,
+                                        vtkInformationVector *outputVector) {
+  ttk::Timer tm{};
+
+  auto *input = vtkTable::GetData(inputVector[0]);
+  auto *output = vtkTable::GetData(outputVector);
+  output->ShallowCopy(input);
+
+  if(SelectFieldsWithRegexp) {
+    // select all input columns whose name is matching the regexp
+    ScalarFields.clear();
+    const auto n = input->GetNumberOfColumns();
+    for(int i = 0; i < n; ++i) {
+      const auto &name = input->GetColumnName(i);
+      if(std::regex_match(name, std::regex(RegexpString))) {
+        ScalarFields.emplace_back(name);
+      }
+    }
+  }
+
+  const auto numberOfRows = input->GetNumberOfRows();
+  const auto numberOfColumns = ScalarFields.size();
+
+  if(numberOfRows <= 0 || numberOfColumns <= 0) {
+    this->printErr("Input matrix has invalid dimensions (rows: "
+                   + std::to_string(numberOfRows)
+                   + ", columns: " + std::to_string(numberOfColumns) + ")");
+    return 0;
+  }
+
+  std::vector<vtkAbstractArray *> arrays{};
+  for(const auto &s : ScalarFields) {
+    arrays.push_back(input->GetColumnByName(s.data()));
+  }
+
+  std::vector<std::vector<double>> inputMatrix(numberOfRows);
+#ifdef TTK_ENABLE_OPENMP
+#pragma omp parallel for num_threads(this->threadNumber_)
+#endif // TTK_ENABLE_OPENMP
+  for(int i = 0; i < numberOfRows; ++i) {
+    for(size_t j = 0; j < numberOfColumns; ++j) {
+      inputMatrix[i].emplace_back(arrays[j]->GetVariantValue(i).ToDouble());
```

Comment:
Note for later @pierre-guillou 
anyway to pass pointers to the raw data down to the base code (like we do for the triangulation?)

Reply:
The `lDistanceMatrix`/`lDistance` modules work on coordinates pointers: one pointer for each point, but the vtkTable arrays are accessed by column. This copy of the input data also serves to transpose the coordinates array to feed it to base modules (while taking care of typing by converting everything to double).
The same thing happens in `ttkDimensionReduction` (from which this snippet is taken).

Reply:
Makes sense, thanks!